llm/prompt.go

```diff
@@ -68,27 +68,58 @@ type Response struct {
 }
 
 func getProvider(ctx context.Context) LLMProvider {
-	if key := os.Getenv("GEMINI_API_KEY"); key != "" {
-		provider := NewGeminiProvider(key, ctx)
-		if provider != nil {
-			return provider
-		}
+	// Define provider configurations
+	providerConfigs := map[string]struct {
+		envKey string
+		initFn func(string, context.Context) LLMProvider
+	}{
+		"gemini": {
+			envKey: "GEMINI_API_KEY",
+			initFn: func(key string, ctx context.Context) LLMProvider {
+				return NewGeminiProvider(key, ctx)
```

Comment:
In this case can't we just initFunc: NewGeminiProvider if args are the same and not wrap with an extra func?

If we can, then maybe just add c ctx arg to others as _ and we can assume this is an interface?